backend/src/test/java/de/kathahuth/gymast/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,78 @@
+package de.kathahuth.gymast.controller;
+
+import de.kathahuth.gymast.db.CategoryMongoDb;
+import de.kathahuth.gymast.model.Category;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.*;
+
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CategoryControllerTest {
+
+    @LocalServerPort
+    private int port;
+
+    private String getUrl() {
+        return "http://localhost:" + port + "/api/category";
+    }
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @Autowired
+    private CategoryMongoDb categoryMongoDb;
+
+    @BeforeEach
+    public void setup() {
+        categoryMongoDb.deleteAll();
+    }
+
+    private Category createPushUpCategory() {
+        return Category.builder()
+                .id("push-up-category")
+                .name("Push Up Category")
+                .categoryNames(List.of("Push Up"))
```

Comment:
Regarding `childrenNames` cut with `categories`: here you use `"Push Up"` which is - i hope so - no available children name :)